scripts/o3de/o3de/manifest.py

```diff
@@ -331,9 +325,16 @@ def get_gem_gems(gem_path: pathlib.Path) -> list:
 def get_gem_external_subdirectories(gem_path: pathlib.Path) -> list:
     gem_object = get_gem_json_data(gem_path=gem_path)
     if gem_object:
-        return list(map(lambda rel_path: (pathlib.Path(gem_path) / rel_path).as_posix(),
-                        gem_object[
-                            'external_subdirectories'])) if 'external_subdirectories' in gem_object else []
+        external_subdirectories = list(map(lambda rel_path: (pathlib.Path(gem_path) / rel_path).as_posix(),
+            gem_object['external_subdirectories'])) if 'external_subdirectories' in gem_object else []
+
+        # recurse into gem subdirectories 
+        for external_subdirectory in external_subdirectories:
+            gem_json_path = pathlib.Path(external_subdirectory).resolve() / 'gem.json'
+            if gem_json_path.is_file():
+                external_subdirectories.extend(get_gem_external_subdirectories(external_subdirectory))
+
+        return external_subdirectories
```

Comment:
We should probably protect against an infinite loop if there is a cycle within the external_subdirectories entries, by adding a visited_set parameter to `get_gem_external_subdirectories`.

```python
def get_gem_external_subdirectories(gem_path: pathlib.Path, visited_gem_paths: set): -> list:
    if gem_path in visited_gem_paths:
        logger.error(f'A cycle has been detected when visiting external subdirectories at gem path "{gem_path}". The visited paths are: {visited_gem_paths}')
        return
    visited_gem_paths.add(gem_path)

    gem_object = get_gem_json_data(gem_path=gem_path)
    if gem_object:
```